scripts/ci/noema_review_gate.py

```
@@ -68,21 +78,115 @@ def run(args: Sequence[str], *, stdin: str | None = None) -> str:
-        stderr=subprocess.PIPE,
-        check=False,
-        shell=False,
+    argv = list(args)
+    last_stderr = ""
+    last_returncode = 1
+    attempts = 1
+    if argv and argv[0] == "gh":
+        attempts = max(1, GH_TRANSIENT_RETRY_ATTEMPTS)
+    attempt = 1
+    while True:
+        completed = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)